apis/storage/v1alpha3/bucketpolicy_types.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
```

Comment:
I think it would be a good idea to introduce this new type at `v1alpha1`. That being said, it has sort of pseudo-existed as part of `Bucket` so I could be convinced otherwise